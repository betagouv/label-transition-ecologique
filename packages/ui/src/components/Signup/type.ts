import {SignupState} from './Signup';

export type SignupView = 'etape1' | 'etape2' | 'infos';

export type SignupDataStep1 = {
  collectivite_id: number;
  email: string;
  password: string;
  cgu_acceptees: boolean;
  collectivite_engagee?: boolean;
};

export type SignupDataStep2 = {
  nom: string;
  prenom: string;
  tel: string;
  role: UserRole;
  poste?: string;
};

export type SignupData = SignupDataStep1 & SignupDataStep2;

export type UserRole =
  | 'equipe_politique'
  | 'services_techniques'
  | 'partenaire'
  | 'bureau_etudes';

export type SignupProps = {
  /** Vue sélectionnée par défaut */
  defaultView?: SignupView;
  /** Valeurs par défaut du formulaire (étape 1) */
  defaultValues?: SignupDataStep1;
  /** Indique que l'option "avec mot de passe" est activée */
  withPassword?: boolean;
  /** Indique qu'un appel réseau est en cours */
  isLoading?: boolean;
  /** Liste de collectivités auxquelles le compte peut être rattaché */
  collectivites: Array<{value: number; label: string}>;
  /** Appelé pour filtrer la liste des collectivités */
  onFilterCollectivites: (value: string) => void;
  /** Appelé à l'envoi du formulaire */
  onSubmit: (data: SignupData) => void;
  /** Appelé à l'annulation du formulaire */
  onCancel: () => void;
};

export type SignupPropsWithState = SignupProps & {signupState: SignupState};
