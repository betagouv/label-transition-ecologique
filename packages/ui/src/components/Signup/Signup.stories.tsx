import {Meta, StoryObj} from '@storybook/react';
import {action} from '@storybook/addon-actions';
import {Signup} from './Signup';

const meta: Meta<typeof Signup> = {
  component: Signup,
  args: {
    collectivites: [
      {value: 1270, label: 'Grenoble'},
      {value: 5460, label: '#Collectivit√© Test'},
    ],
    onSubmit: action('onSubmit'),
    onCancel: action('onCancel'),
    onFilterCollectivites: action('onFilterCollectivites'),
  },
};

export default meta;

type Story = StoryObj<typeof Signup>;

export const Default: Story = {
  args: {
    defaultValues: {
      collectivite_id: 1270,
      email: 'yolo@dodo.com',
      cgu_acceptees: true,
      password: '',
    },
  },
};

export const Etape1AvecMdp: Story = {
  args: {
    withPassword: true,
    defaultView: 'etape1',
  },
};

export const SansSelectionCollectivite: Story = {
  args: {
    noCollectivite: true,
    defaultView: 'etape1',
  },
};

export const Etape2: Story = {
  args: {
    defaultView: 'etape2',
  },
};
