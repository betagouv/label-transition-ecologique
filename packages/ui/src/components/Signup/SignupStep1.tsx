import {Controller} from 'react-hook-form';
import {useForm} from 'react-hook-form';
import {zodResolver} from '@hookform/resolvers/zod';
import {z} from 'zod';
import {Tab, Tabs} from '@design-system/Tabs';
import {ModalFooterOKCancel} from '@design-system/Modal';
import {Field} from '@design-system/Field';
import {Input} from '@design-system/Input';
import {Checkbox} from '@design-system/Checkbox';
import {Select} from '@design-system/Select';
import {Button} from '@design-system/Button';
import {SignupDataStep1, SignupPropsWithState} from './type';
import {useState} from 'react';

const URL_CGU = 'https://territoiresentransitions.fr/cgu';
const URL_DCP = 'https://www.ademe.fr/donnees-personnelles/';

/** Gestionnaire d'état pour le formulaire de l'étape 1 */
const useSignupStep1 = (
  isPasswordless: boolean,
  defaultValues: SignupDataStep1
) => {
  const validationSchema = z.object({
    collectivite_id: z.number({
      invalid_type_error: 'Sélectionnez une collectivité',
    }),
    email: z.string().email({message: 'Un email valide est requis'}),
    password: z
      .string()
      .refine(value => (isPasswordless ? true : value.length >= 8), {
        message: 'Le mot de passe doit comporter au moins 8 caractères',
      }),
    cgu_acceptees: z.boolean().refine(value => !!value, {
      message: "Vous devez accepter le cadre d'utilisation de la plateforme",
    }),
    collectivite_engagee: z.boolean().optional(),
  });

  return useForm({
    reValidateMode: 'onChange',
    resolver: zodResolver(validationSchema),
    defaultValues: defaultValues || {
      collectivite_id: null,
      password: '',
      email: '',
      cgu_acceptees: false,
      collectivite_engagee: false,
    },
  });
};

/**
 * Affiche l'étape 1 du panneau de création de compte
 */
export const SignupStep1 = (props: SignupPropsWithState) => {
  const {signupState, withPassword} = props;
  const {formData} = signupState;
  const [isPasswordless, setIsPasswordless] = useState(!withPassword);
  const form = useSignupStep1(isPasswordless, formData);

  return (
    <Tabs
      className="justify-center"
      defaultActiveTab={isPasswordless ? 0 : 1}
      onChange={activeTab => {
        if (activeTab === 0) {
          // reset le champ mdp qui peut être rempli quand on passe d'un onglet à l'autre
          form.setValue('password', '');
          setIsPasswordless(true);
        } else {
          setIsPasswordless(false);
        }
      }}
    >
      <Tab label="Compte sans mot de passe">
        <SignupStep1Form {...props} form={form} isPasswordless />
      </Tab>
      <Tab label="Compte avec mot de passe">
        <SignupStep1Form {...props} form={form} />
      </Tab>
    </Tabs>
  );
};

/**
 * Affiche le formulaire pour l'étape 1 de la création de compte
 */
const SignupStep1Form = (
  props: SignupPropsWithState & {
    form: ReturnType<typeof useSignupStep1>;
    isPasswordless?: boolean;
  }
) => {
  const {
    isLoading,
    isPasswordless,
    collectivites,
    onFilterCollectivites,
    onCancel,
    form,
    signupState: {setFormData, setView},
  } = props;
  const {
    handleSubmit,
    control,
    register,
    setValue,
    formState: {isValid, errors},
  } = form;

  // enregistre les valeurs et passe à l'étape suivante
  const onSubmit = data => {
    setFormData(data);
    setView('etape2');
  };

  return (
    <form className="flex flex-col gap-4" onSubmit={handleSubmit(onSubmit)}>
      <Field
        title="Nom de la collectivité"
        state={errors.collectivite_id ? 'error' : undefined}
        message={errors.collectivite_id?.message.toString()}
      >
        <Controller
          name="collectivite_id"
          control={control}
          render={({field}) => {
            return (
              <Select
                debounce={500}
                options={collectivites}
                values={field.value ? [field.value] : undefined}
                isSearcheable
                onSearch={onFilterCollectivites}
                disabled={isLoading}
                onChange={value => {
                  setValue(
                    field.name,
                    field.value === value ? null : (value as number),
                    {
                      shouldValidate: true,
                    }
                  );
                }}
              />
            );
          }}
        />
      </Field>
      <Field
        title="Email de connexion"
        htmlFor="email"
        state={errors.email ? 'error' : undefined}
        message={errors.email?.message.toString()}
      >
        <Input id="email" type="text" {...register('email')} />
      </Field>
      {!isPasswordless && (
        <Field
          title="Mot de passe"
          htmlFor="password"
          state={errors.password ? 'error' : undefined}
          message={errors.password?.message.toString()}
        >
          <Input id="password" type="password" {...register('password')} />
        </Field>
      )}
      <hr className="mt-5" />
      <Checkbox
        className="font-medium"
        label="J’accepte le cadre d’utilisation de la plateforme (obligatoire)"
        message={
          <>
            <Button variant="underlined" size="sm" href={URL_CGU}>
              Les conditions générales
            </Button>
            <span className="text-sm mx-2">|</span>
            <Button variant="underlined" size="sm" href={URL_DCP}>
              la politique de protection des données de l'ADEME
            </Button>
          </>
        }
        {...register('cgu_acceptees')}
      />
      <Checkbox
        label="Collectivité engagée dans le programme Territoire Engagé Transition Ecologique"
        {...register('collectivite_engagee')}
      />
      <ModalFooterOKCancel
        btnCancelProps={{onClick: onCancel}}
        btnOKProps={{
          onClick: handleSubmit(onSubmit),
          disabled: !isValid,
          children: "Passer à l'étape 2",
          icon: 'arrow-right-line',
          iconPosition: 'right',
        }}
      />
    </form>
  );
};
