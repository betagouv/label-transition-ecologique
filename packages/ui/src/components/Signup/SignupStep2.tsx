import {Controller, useForm} from 'react-hook-form';
import {zodResolver} from '@hookform/resolvers/zod';
import {z} from 'zod';
import {ModalFooterOKCancel} from '@design-system/Modal';
import {Field} from '@design-system/Field';
import {Input} from '@design-system/Input';
import {Select} from '@design-system/Select';
import {SignupPropsWithState, UserRole} from './type';
import {FormSectionGrid} from '@design-system/FormSection';

const ROLES: Array<{value: UserRole; label: string}> = [
  {value: 'equipe_politique', label: 'Équipe politique'},
  {value: 'services_techniques', label: 'Directions et services techniques'},
  {value: 'partenaire', label: 'Partenaire'},
  {value: 'bureau_etudes', label: "Bureau d'études"},
];

/** Gestionnaire d'état pour le formulaire de l'étape 2 */
const useSignupStep2 = () => {
  const validationSchema = z.object({
    nom: z.string().min(1, 'Champ requis'),
    prenom: z.string().min(1, 'Champ requis'),
    tel: z.string().min(1, 'Champ requis'),
    role: z
      .string({invalid_type_error: 'Sélectionnez un rôle'})
      .min(1, 'Champ requis'),
    poste: z.string().optional(),
  });

  return useForm({
    resolver: zodResolver(validationSchema),
  });
};

/**
 * Affiche l'étape 2 du panneau de création de compte
 */
export const SignupStep2 = (props: SignupPropsWithState) => {
  const {
    handleSubmit,
    control,
    register,
    setValue,
    formState: {isValid, errors},
  } = useSignupStep2();

  const {
    onCancel,
    onSubmit,
    signupState: {formData},
  } = props;

  const onSubmitForm = data => {
    onSubmit({...formData, ...data});
  };

  return (
    <form className="flex flex-col gap-4" onSubmit={handleSubmit(onSubmitForm)}>
      <FormSectionGrid>
        <Field
          title="Nom"
          htmlFor="nom"
          state={errors.nom ? 'error' : undefined}
          message={errors.nom?.message.toString()}
        >
          <Input id="nom" type="text" {...register('nom')} />
        </Field>
        <Field
          title="Prénom"
          htmlFor="prenom"
          state={errors.prenom ? 'error' : undefined}
          message={errors.prenom?.message.toString()}
        >
          <Input id="prenom" type="text" {...register('prenom')} />
        </Field>
        <Field
          className="md:col-span-2"
          title="Numéro de téléphone"
          htmlFor="tel"
          state={errors.tel ? 'error' : 'info'}
          message={
            errors.tel?.message.toString() ||
            "L'accès à votre numéro est réservé à l'équipe ADEME de l'outil et le contact par mail restera privilégié"
          }
        >
          <Input id="tel" type="text" {...register('tel')} />
        </Field>
        <Field
          title="Rôle"
          state={errors.role ? 'error' : undefined}
          message={errors.role?.message.toString()}
        >
          <Controller
            name="role"
            control={control}
            render={({field}) => (
              <Select
                options={ROLES}
                values={field.value ? [field.value] : undefined}
                onChange={value => {
                  setValue(
                    field.name,
                    field.value === value ? null : (value as string),
                    {
                      shouldValidate: true,
                    }
                  );
                }}
              />
            )}
          />
        </Field>
        <Field
          title="Intitulé de poste"
          htmlFor="poste"
          state={errors.poste ? 'error' : undefined}
          message={errors.poste?.message.toString()}
        >
          <Input id="poste" type="text" {...register('poste')} />
        </Field>
      </FormSectionGrid>
      <ModalFooterOKCancel
        btnCancelProps={{onClick: onCancel}}
        btnOKProps={{
          onClick: handleSubmit(onSubmitForm),
          disabled: !isValid,
        }}
      />
    </form>
  );
};
