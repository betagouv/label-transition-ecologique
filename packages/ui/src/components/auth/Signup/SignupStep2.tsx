import {SignupPropsWithState} from './type';
import {VerifyOTP} from '@components/auth/VerifyOTP/VerifyOTP';

/**
 * Affiche l'étape 2 du panneau de création de compte
 */
export const SignupStep2 = (props: SignupPropsWithState) => {
  const {
    defaultValues,
    isLoading,
    error,
    onCancel,
    onSubmit,
    onResend,
    formState: {email},
  } = props;

  return (
    <VerifyOTP
      type="signup"
      isLoading={isLoading}
      defaultValues={{
        email: defaultValues?.email || email,
        otp: defaultValues?.otp,
      }}
      error={error}
      onCancel={onCancel}
      onSubmit={onSubmit}
      onResend={onResend}
    />
  );
};
