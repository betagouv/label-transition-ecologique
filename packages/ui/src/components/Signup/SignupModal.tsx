import {useState} from 'react';
import {Modal} from '@design-system/Modal';
import {SignupProps} from './type';
import {Signup} from './Signup';

/**
 * Encapsule le panneau de crÃ©ation de compte dans une modale
 */
export const SignupModal = (props: SignupProps) => {
  const [isOpen, setIsOpen] = useState(true);
  const onClose = () => {
    setIsOpen(false);
    props.onCancel();
  };

  return (
    <Modal
      backdropBlur
      size="lg"
      openState={{isOpen, setIsOpen}}
      onClose={onClose}
      render={() => <Signup {...props} onCancel={() => onClose()} />}
    />
  );
};
