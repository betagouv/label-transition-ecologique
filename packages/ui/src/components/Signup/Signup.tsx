import {SignupProps, SignupView} from './type';
import {SignupStep1} from './SignupStep1';
import {SignupStep2} from '@components/Signup/SignupStep2';
import {useState, useEffect} from 'react';

/**
 * Affiche le panneau création de compte
 */
export const Signup = (props: SignupProps) => {
  const signupState = useSignupState(props);
  const {view} = signupState;

  if (view === 'etape1') {
    return <SignupStep1 {...props} signupState={signupState} />;
  }

  if (view === 'etape2') {
    return <SignupStep2 {...props} signupState={signupState} />;
  }
};

/**
 * Gère l'état du formulaire de création de compte
 */
const useSignupState = (props: SignupProps) => {
  const {defaultView = 'etape1', defaultValues} = props;

  // la vue courante
  const [view, setView] = useState<SignupView>(defaultView);

  // synchronise l'état interne avec la vue voulue
  useEffect(() => setView(defaultView), [defaultView]);

  // données collectées dans les formulaires
  const [formData, setFormData] = useState(defaultValues);

  return {
    view,
    setView,
    formData,
    setFormData,
  };
};
export type SignupState = ReturnType<typeof useSignupState>;
