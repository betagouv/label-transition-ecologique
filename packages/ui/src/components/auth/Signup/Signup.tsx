import {MailSendMessage} from '@components/auth/Login/MailSendMessage';
import {useFormState} from '@components/auth/Login/useFormState';
import {SignupProps} from './type';
import {SignupStep1} from './SignupStep1';
import {SignupStep2} from './SignupStep2';
import {SignupStep3} from './SignupStep3';
import {SignupStep4} from './SignupStep4';

/**
 * Affiche le panneau création de compte
 */
export const Signup = (props: SignupProps) => {
  const formState = useFormState(props);
  const {view} = props;

  if (view === 'etape1') {
    return <SignupStep1 {...props} formState={formState} />;
  }

  if (view === 'etape2') {
    return <SignupStep2 {...props} formState={formState} />;
  }

  if (view === 'etape3') {
    return <SignupStep3 {...props} formState={formState} />;
  }

  if (view === 'etape4') {
    return <SignupStep4 {...props} formState={formState} />;
  }

  if (view === 'msg_lien_envoye') {
    return (
      <MailSendMessage
        data-test="lien-envoye"
        message1="Pour activer votre compte, veuillez consulter votre boite mail et"
        message2="suivre le lien reçu !"
      />
    );
  }
};
