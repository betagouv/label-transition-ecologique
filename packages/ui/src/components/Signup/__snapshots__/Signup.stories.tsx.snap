// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Signup Default smoke-test 1`] = `
<div class="overflow-hidden">
  <ul class="inline-flex rounded-lg bg-grey-2 p-2 gap-3 w-full overflow-x-auto justify-center"
      role="tablist"
  >
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md border border-grey-3 rounded-md shadow-button bg-white text-primary-9"
              type="button"
              role="tab"
              id="tab-0"
              aria-selected="true"
      >
        Compte sans mot de passe
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md text-primary-10"
              type="button"
              role="tab"
              id="tab-1"
              aria-selected="false"
      >
        Compte avec mot de passe
      </button>
    </li>
  </ul>
  <div id="tabpanel-0"
       class
       role="tabpanel"
       aria-labelledby="tab-0"
       tabindex="0"
       label="Compte sans mot de passe"
  >
    <form class="flex flex-col gap-4">
      <div class="flex-grow flex flex-col gap-3">
        <label>
          <div class="font-medium text-grey-8">
            Nom de la collectivité
          </div>
        </label>
        <button aria-expanded="false"
                aria-label="ouvrir le menu"
                class="w-full rounded-xl border border-solid border-grey-4 disabled:border-grey-3 bg-grey-1 hover:!bg-primary-0 disabled:hover:!bg-grey-1 overflow-hidden"
                type="button"
        >
          <div class="flex px-4 min-h-[3rem] py-2">
            <div class="flex grow flex-wrap gap-2 mr-4">
              <div class="flex items-center gap-2 grow">
                <div data-test="Badge-undefined"
                     class="bg-grey-1 border-grey-5 flex items-center gap-1 max-w-max px-3 py-1 border border-solid rounded"
                >
                  <span class="text-grey-8 font-bold uppercase leading-4 text-left line-clamp-1 mt-0.5 text-sm">
                    Grenoble
                  </span>
                  <div class="flex rounded-full cursor-pointer">
                    <span class="ri-close-circle-fill font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] text-primary-7">
                    </span>
                  </div>
                </div>
              </div>
              <input type="text"
                     class="w-full text-sm outline-0 placeholder:text-grey-6 py-1"
                     placeholder="Rechercher par mots-clés"
                     value
              >
            </div>
            <span class="ri-arrow-down-s-line font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] mt-2 ml-auto text-primary-9">
            </span>
          </div>
        </button>
      </div>
      <div class="flex-grow flex flex-col gap-3">
        <label for="email">
          <div class="font-medium text-grey-8">
            Email de connexion
          </div>
        </label>
        <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
          <input type="text"
                 class="grow text-grey-8 px-4 outline-none text-md py-3"
                 id="email"
                 name="email"
          >
        </div>
      </div>
      <hr class="mt-5">
      <div class="flex gap-2 min-w-min">
        <input id=":r1:"
               type="checkbox"
               class="!appearance-none shrink-0 checked:before:icon-check-line checked:bg-primary checked:disabled:bg-primary-5 flex content-center justify-center mt-0.5 border border-solid !border-grey-6 rounded flex-wrap w-5 h-5 !text-white  checked:!border-transparent checked:hover:bg-primary-8 checked:!disabled:border-grey-4 font-medium"
               name="cgu_acceptees"
        >
        <div class="flex flex-col gap-1">
          <label for=":r1:"
                 class="inline-flex items-center cursor-pointer font-medium text-grey-8"
          >
            J’accepte le cadre d’utilisation de la plateforme (obligatoire)
          </label>
          <div class="flex items-center text-grey-7 my-auto">
            <a href="https://territoiresentransitions.fr/cgu"
               class="bg-none after:hidden w-fit flex items-center border-solid group gap-2 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-sm py-2.5 px-6"
            >
              Les conditions générales
            </a>
            <span class="text-sm mx-2">
              |
            </span>
            <a href="https://www.ademe.fr/donnees-personnelles/"
               class="bg-none after:hidden w-fit flex items-center border-solid group gap-2 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-sm py-2.5 px-6"
            >
              la politique de protection des données de l'ADEME
            </a>
          </div>
        </div>
      </div>
      <div class="flex gap-2 min-w-min">
        <input id=":r2:"
               type="checkbox"
               class="!appearance-none shrink-0 checked:before:icon-check-line checked:bg-primary checked:disabled:bg-primary-5 flex content-center justify-center mt-0.5 border border-solid !border-grey-6 rounded flex-wrap w-5 h-5 !text-white  checked:!border-transparent checked:hover:bg-primary-8 checked:!disabled:border-grey-4"
               name="collectivite_engagee"
        >
        <div class="flex flex-col gap-1">
          <label for=":r2:"
                 class="inline-flex items-center cursor-pointer font-medium text-grey-8"
          >
            Collectivité engagée dans le programme Territoire Engagé Transition Ecologique
          </label>
        </div>
      </div>
      <hr class="mt-4">
      <div class="flex gap-4 justify-end">
        <button type="button"
                class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
        >
          Annuler
        </button>
        <button type="submit"
                class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 text-base py-3 px-6 flex-row-reverse"
        >
          <span class="ri-arrow-right-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-white group-hover:fill-white">
          </span>
          Passer à l'étape 2
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`components/Signup Etape1AvecMdp smoke-test 1`] = `
<div class="overflow-hidden">
  <ul class="inline-flex rounded-lg bg-grey-2 p-2 gap-3 w-full overflow-x-auto justify-center"
      role="tablist"
  >
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md text-primary-10"
              type="button"
              role="tab"
              id="tab-0"
              aria-selected="false"
      >
        Compte sans mot de passe
      </button>
    </li>
    <li role="presentation"
        class="p-0"
    >
      <button class="px-3 py-1 font-bold w-max hover:rounded-md hover:shadow-button hover:!bg-primary-2 hover:text-primary-9 text-md border border-grey-3 rounded-md shadow-button bg-white text-primary-9"
              type="button"
              role="tab"
              id="tab-1"
              aria-selected="true"
      >
        Compte avec mot de passe
      </button>
    </li>
  </ul>
  <div id="tabpanel-1"
       class
       role="tabpanel"
       aria-labelledby="tab-1"
       tabindex="1"
       label="Compte avec mot de passe"
  >
    <form class="flex flex-col gap-4">
      <div class="flex-grow flex flex-col gap-3">
        <label>
          <div class="font-medium text-grey-8">
            Nom de la collectivité
          </div>
        </label>
        <button aria-expanded="false"
                aria-label="ouvrir le menu"
                class="w-full rounded-xl border border-solid border-grey-4 disabled:border-grey-3 bg-grey-1 hover:!bg-primary-0 disabled:hover:!bg-grey-1 overflow-hidden"
                type="button"
        >
          <div class="flex px-4 min-h-[3rem] py-2">
            <div class="flex grow flex-wrap gap-2 mr-4">
              <input type="text"
                     class="w-full text-sm outline-0 placeholder:text-grey-6"
                     placeholder="Rechercher par mots-clés"
                     value
              >
            </div>
            <span class="ri-arrow-down-s-line font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] mt-2 ml-auto text-primary-9">
            </span>
          </div>
        </button>
      </div>
      <div class="flex-grow flex flex-col gap-3">
        <label for="email">
          <div class="font-medium text-grey-8">
            Email de connexion
          </div>
        </label>
        <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
          <input type="text"
                 class="grow text-grey-8 px-4 outline-none text-md py-3"
                 id="email"
                 name="email"
          >
        </div>
      </div>
      <div class="flex-grow flex flex-col gap-3">
        <label for="password">
          <div class="font-medium text-grey-8">
            Mot de passe
          </div>
        </label>
        <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
          <input type="password"
                 class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4"
                 id="password"
                 name="password"
          >
          <button title="Voir le mot de passe"
                  class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
          >
            <span class="ri-eye-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
            </span>
          </button>
        </div>
      </div>
      <hr class="mt-5">
      <div class="flex gap-2 min-w-min">
        <input id=":r4:"
               type="checkbox"
               class="!appearance-none shrink-0 checked:before:icon-check-line checked:bg-primary checked:disabled:bg-primary-5 flex content-center justify-center mt-0.5 border border-solid !border-grey-6 rounded flex-wrap w-5 h-5 !text-white  checked:!border-transparent checked:hover:bg-primary-8 checked:!disabled:border-grey-4 font-medium"
               name="cgu_acceptees"
        >
        <div class="flex flex-col gap-1">
          <label for=":r4:"
                 class="inline-flex items-center cursor-pointer font-medium text-grey-8"
          >
            J’accepte le cadre d’utilisation de la plateforme (obligatoire)
          </label>
          <div class="flex items-center text-grey-7 my-auto">
            <a href="https://territoiresentransitions.fr/cgu"
               class="bg-none after:hidden w-fit flex items-center border-solid group gap-2 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-sm py-2.5 px-6"
            >
              Les conditions générales
            </a>
            <span class="text-sm mx-2">
              |
            </span>
            <a href="https://www.ademe.fr/donnees-personnelles/"
               class="bg-none after:hidden w-fit flex items-center border-solid group gap-2 cursor-pointer border-b border-t-0 border-x-0 !p-px font-medium hover:border-b-2 hover:!pb-0 text-primary-8 !bg-transparent border-primary-7 text-sm py-2.5 px-6"
            >
              la politique de protection des données de l'ADEME
            </a>
          </div>
        </div>
      </div>
      <div class="flex gap-2 min-w-min">
        <input id=":r5:"
               type="checkbox"
               class="!appearance-none shrink-0 checked:before:icon-check-line checked:bg-primary checked:disabled:bg-primary-5 flex content-center justify-center mt-0.5 border border-solid !border-grey-6 rounded flex-wrap w-5 h-5 !text-white  checked:!border-transparent checked:hover:bg-primary-8 checked:!disabled:border-grey-4"
               name="collectivite_engagee"
        >
        <div class="flex flex-col gap-1">
          <label for=":r5:"
                 class="inline-flex items-center cursor-pointer font-medium text-grey-8"
          >
            Collectivité engagée dans le programme Territoire Engagé Transition Ecologique
          </label>
        </div>
      </div>
      <hr class="mt-4">
      <div class="flex gap-4 justify-end">
        <button type="button"
                class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
        >
          Annuler
        </button>
        <button type="submit"
                disabled
                class="w-fit flex items-center border-solid group gap-2 cursor-not-allowed rounded-lg border font-bold !text-primary-3 !bg-primary-4 !border-primary-4 text-base py-3 px-6 flex-row-reverse"
        >
          <span class="ri-arrow-right-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-3">
          </span>
          Passer à l'étape 2
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`components/Signup Etape2 smoke-test 1`] = `
<form class="flex flex-col gap-4">
  <div class="grid grid-cols-1 md:grid-cols-2 gap-6">
    <div class="flex-grow flex flex-col gap-3">
      <label for="nom">
        <div class="font-medium text-grey-8">
          Nom
        </div>
      </label>
      <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
        <input type="text"
               class="grow text-grey-8 px-4 outline-none text-md py-3"
               id="nom"
               name="nom"
        >
      </div>
    </div>
    <div class="flex-grow flex flex-col gap-3">
      <label for="prenom">
        <div class="font-medium text-grey-8">
          Prénom
        </div>
      </label>
      <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
        <input type="text"
               class="grow text-grey-8 px-4 outline-none text-md py-3"
               id="prenom"
               name="prenom"
        >
      </div>
    </div>
    <div class="flex-grow flex flex-col gap-3 md:col-span-2">
      <label for="tel">
        <div class="font-medium text-grey-8">
          Numéro de téléphone
        </div>
      </label>
      <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
        <input type="text"
               class="grow text-grey-8 px-4 outline-none text-md py-3"
               id="tel"
               name="tel"
        >
      </div>
      <div class="flex items-center text-info-1">
        <span class="ri-information-fill font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] mr-1">
        </span>
        <span class="text-xs">
          L'accès à votre numéro est réservé à l'équipe ADEME de l'outil et le contact par mail restera privilégié
        </span>
      </div>
    </div>
    <div class="flex-grow flex flex-col gap-3">
      <label>
        <div class="font-medium text-grey-8">
          Rôle
        </div>
      </label>
      <button aria-expanded="false"
              aria-label="ouvrir le menu"
              class="w-full rounded-xl border border-solid border-grey-4 disabled:border-grey-3 bg-grey-1 hover:!bg-primary-0 disabled:hover:!bg-grey-1 overflow-hidden"
              type="button"
      >
        <div class="flex px-4 min-h-[3rem] py-2">
          <div class="flex grow flex-wrap gap-2 mr-4">
            <span class="my-auto text-left text-xs text-grey-6 line-clamp-1">
              Sélectionner une option
            </span>
          </div>
          <span class="ri-arrow-down-s-line font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] mt-2 ml-auto text-primary-9">
          </span>
        </div>
      </button>
    </div>
    <div class="flex-grow flex flex-col gap-3">
      <label for="poste">
        <div class="font-medium text-grey-8">
          Intitulé de poste
        </div>
      </label>
      <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
        <input type="text"
               class="grow text-grey-8 px-4 outline-none text-md py-3"
               id="poste"
               name="poste"
        >
      </div>
    </div>
  </div>
  <hr class="mt-4">
  <div class="flex gap-4 justify-end">
    <button type="button"
            class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
    >
      Annuler
    </button>
    <button type="submit"
            disabled
            class="w-fit flex items-center border-solid group gap-2 cursor-not-allowed rounded-lg border font-bold !text-primary-3 !bg-primary-4 !border-primary-4 text-base py-3 px-6"
    >
      Valider
    </button>
  </div>
</form>
`;
